### PR TITLE
Hexen: brightmaps

### PR DESCRIPTION
**Pull request by JNechaevsky**
Hello.
* Wall textures: [screenshot1](https://user-images.githubusercontent.com/21193394/170111310-cc19db51-9f49-4cb6-a57e-c1c6ebb612d4.png) and [screenshot2](https://user-images.githubusercontent.com/21193394/170111352-65f545f6-066d-4b65-994f-d0f6d3f82293.png)
* Midtextures on two-sided walls: [screenshot](https://user-images.githubusercontent.com/21193394/170111552-629c762c-bb82-4d5b-ba65-2b7b0c122965.png)
* Things/thinkers/actors/mobjs: [screenshot](https://user-images.githubusercontent.com/21193394/170111485-2f4c28e5-6833-4ff2-9985-4e22dfe515cd.png)
* These gentlemen: [screenshot1](https://user-images.githubusercontent.com/21193394/170111639-98a50fb2-55b6-4626-b8dc-57e817f743ec.png) and [screenshot2](https://user-images.githubusercontent.com/21193394/170112101-8f13c690-5a4c-4ca8-bfbe-74ec99138958.png)
* STbar weapons: [Timon's Axe](https://user-images.githubusercontent.com/21193394/170112504-a9ae19e9-bc95-47e0-9fa8-ff274d146680.png), [Quietus](https://user-images.githubusercontent.com/21193394/170112633-07cd10d0-0a3c-4d8b-a404-8013afbb4e7f.png), [Serpent Staff](https://user-images.githubusercontent.com/21193394/170112839-fa8b6811-2450-477a-85d6-6ed60558fade.png), [Icicles](https://user-images.githubusercontent.com/21193394/170113218-3d2f93ac-abb7-40dd-ba8c-346b807d4b25.png), [Arc of Death](https://user-images.githubusercontent.com/21193394/170113307-b846b0bd-4468-473c-beed-539a91f14e4d.png) and [Bloodscourge](https://user-images.githubusercontent.com/21193394/170113443-f96bdf99-01fb-4087-bcff-07c48cfea12d.png)

Few objects didn't make the final cut because of shared pixels on flame on objects themselves. 
No brightmaps for floors because no appropriate floor textures.
No half-brights, sacrificed even in Inter-Hexen for simplicity.

Testing map: [hx_brightmaps.zip](https://github.com/fabiangreffrath/crispy-doom/files/8765631/hx_brightmaps.zip)

### CODE REVIEW

File: src/hexen/r_bmaps.c [hunk 1]
```diff
@@ -0,0 +1,710 @@
+//
+// Copyright(C) 1993-1996 Id Software, Inc.
+// Copyright(C) 2005-2014 Simon Howard
+// Copyright(C) 2013-2017 Brad Harding
```
Comment (fabiangreffrath) [inline src/hexen/r_bmaps.c:4]: Your copyright here and in the header (and on Heretic), please.

Comment (JNechaevsky) [reply src/hexen/r_bmaps.c:4]: Done. I've also disabled brightmaps on foggy maps, here's why they shouldn't be applied: [screenshot](https://user-images.githubusercontent.com/21193394/170188315-112fb6da-07e7-459a-abf7-4096a6856f18.png).

**Comment by JNechaevsky** [comment]
I think it's all done, except there is no Crispness menu item yet.
@fabiangreffrath your turn please, need to decide how we should proceed to commit everything correctly with @mikeday0's PR.

**Comment by fabiangreffrath** [comment]
I'll merge @mikeday0 's PR now, so you could please merge the master branch into your PR branch and I'll merge that afterwards.

**Comment by JNechaevsky** [comment]
That's odd, I'm fetching `hx_bmaps` branch, but there is no indication about update necessity, and no merge conflicts either. Could it because of `hx_bmaps` branch is located right in `fabiangreffrath/crispy-doom`, or am I getting something wrong as usual? 🤔

**Comment by fabiangreffrath** [comment]
Well, there are no merge conflicts now. But once you'll attempt to add the menu item for the brightmaps feature, there will be. If you merge the master branch before, you'll be able to work on this part based on the current code, and not the one you used to branch off before.

**Comment by JNechaevsky** [comment]
Ah, understood, I should update working folder as well. Will have to do this on evening, I'm late for work _(as usual!)_.

**Comment by JNechaevsky** [comment]
Done, but mn_menu.c now gives merging conflicts. How should I proceed?

**Comment by JNechaevsky** [comment]
> It looks like you tried to apply the master branch changes manually?

True... Figured out, thanks! Now it should be fine, menu item is fully operational and the rest of job already done.

**Comment by fabiangreffrath** [comment]
Thank you very much once again!

**Comment by JNechaevsky** [comment]
It was a pleasure! Before I move on to make Doom bmaps `const` - do we need brightmaps for mid-textures in Doom and Heretic (https://github.com/fabiangreffrath/crispy-doom/blob/master/src/hexen/r_segs.c#L136-L140)? 

Practically, it's not really needed. The only case I can imagine for non-solid switch is... Fake wall, maybe? Technically, however, this should be okay: no extra calculations are needed, we just adressing to `texturebrightmap` with `texnum` number, the rest is `R_DrawColumn` job, which already getting various `dc_colormap` with different types (brightmap or simply `[0]`).

**Pull request merged**